### PR TITLE
fix: Incorrect type `BaseBuilder::$tableName`

### LINKED ISSUE #4457: Bug: Incorrect type BaseBuilder::$tableName
**Direction**
Property: string 
https://github.com/codeigniter4/CodeIgniter4/blob/8322f3aa396ce91b7e56185cc7eb7c8a0dcf8b56/system/Database/BaseBuilder.php#L163-L165

But in the constructor, you can assign an array to the property. (See docBlock) 
https://github.com/codeigniter4/CodeIgniter4/blob/8322f3aa396ce91b7e56185cc7eb7c8a0dcf8b56/system/Database/BaseBuilder.php#L254-L269

The method returns a string. ( See docBlock and the return typehint)
https://github.com/codeigniter4/CodeIgniter4/blob/8322f3aa396ce91b7e56185cc7eb7c8a0dcf8b56/system/Database/BaseBuilder.php#L317-L321

**Affected module(s)**
BaseBuilder аnd maybe somewhere else

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Fixes #4457
- make sure `$this->tableName` is string

`$this->tableName` is used only in `BaseBuilder::getTable()`.
`BaseBuilder::getTable()` is used only in `Model::builder()`:
https://github.com/codeigniter4/CodeIgniter4/blob/b8fe16d2ff708971693a5e3325b74447ffb93fdd/system/Model.php#L524

`$table` or `$this->table` in the `Model` is a single table name without an alias.
So the above code is just to make sure that `$table` and the Builder's table name are the same or not.
If these are the same table, it returns existing `$this->builder`.

In other words, if the Builder's table name is not a single table, these are always not the same.

Therefore `BaseBuilder::getTable()` does not have to return array or multiple table string like `'tableA, tableB'`.
Returning empty string is enough.

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by MGatner — CHANGES_REQUESTED** [review]
I think this is a fine fix for a messy situation. Definitely need to revisit this in version 5. Please add a note to the changelogs on the off chance someone was using this, and link this PR or issue to our version 5 project column.

**Comment by kenjis** [comment]
@MGatner Is there any chance to include this PR in the next version (4.1.6)?

I created a more aggressive PR #5387 for v5.0.
It's kind of a memo to remember my current thoughts.

Added a note in the project Version 5.

**Comment by MGatner** [comment]
@kenjis I think that is a good approach but I would still like a note in the changelogs, something like this:

> Multiple table names will no longer be stored in `BaseBuilder::$tableName` - an empty string will be used instead.

**Comment by kenjis** [comment]
@MGatner I added in the changelog.

**Review by MGatner — APPROVED** [review]

**Comment by MGatner** [comment]
Thank you!
